src/extension/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java

```diff
@@ -268,8 +278,27 @@ public PipelineBuilder transform(final boolean transformToScreenCoordinates) {
      *
      * @param isTransformToScreenCoordinates Use screen coordinate space simplification tolerance
      */
-    public PipelineBuilder simplify(boolean isTransformToScreenCoordinates) {
+    public PipelineBuilder simplify(
+            boolean isTransformToScreenCoordinates,
+            final Set<RenderingHints.Key> fsHints,
+            final Hints qHints) {
 
+        if (fsHints != null && qHints != null) {
+            // ... if possible we let the datastore do the generalization
+            if (fsHints.contains(Hints.GEOMETRY_SIMPLIFICATION)) {
+                // good, we don't need to perform in memory generalization, the datastore
+                // does it all for us
+                qHints.put(Hints.GEOMETRY_SIMPLIFICATION, context.sourceCRSSimplificationDistance);
+
+                return this;
+
+            } else if (fsHints.contains(Hints.GEOMETRY_DISTANCE)) {
```

Comment:
Not so sure about this (actually copied from other Renderer). in case of GEOMETRY_SIMPLIFICATION do we have to skip the GEOMETRY_DISTANCE hint? why can't we leverage on both?
Note also that with VectorTile GEOMETRY_SIMPLIFICATION is causing invalid geometries as discussed [here](https://sourceforge.net/p/geoserver/mailman/message/37045403/)